test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/ActionOverloadingTests/Tests/ActionOverloadingEndToEndTests.cs

```diff
@@ -0,0 +1,266 @@
+
+        [Fact]
+        public void ExecuteOverloadedOperations()
+        {
+            var product = _context.Execute<Common.Clients.EndToEnd.Product>(new Uri(_baseUri + "Products(-10)", UriKind.Absolute)).Single();
+            OperationDescriptor productOperationDescriptor = _context.GetEntityDescriptor(product).OperationDescriptors.Single();
+
+            Assert.Equal("http://localhost/odata/Products(-10)/Default.RetrieveProduct", productOperationDescriptor.Target.AbsoluteUri, true);
+            Assert.Equal("http://localhost/odata/$metadata#Default.RetrieveProduct", productOperationDescriptor.Metadata.AbsoluteUri, true);
+
+            Common.Clients.EndToEnd.OrderLine orderLine = _context.Execute<Common.Clients.EndToEnd.OrderLine>(new Uri(_baseUri + "OrderLines(OrderId=-10,ProductId=-10)", UriKind.Absolute)).Single();
+
+            OperationDescriptor orderLineOperationDescriptor = _context.GetEntityDescriptor(orderLine).OperationDescriptors.Single();
+            Assert.Equal("http://localhost/odata/OrderLines(OrderId=-10,ProductId=-10)/Default.RetrieveProduct", orderLineOperationDescriptor.Target.AbsoluteUri, true);
+            Assert.Equal("http://localhost/odata/$metadata#Default.RetrieveProduct", orderLineOperationDescriptor.Metadata.AbsoluteUri, true);
+
+            // service operation
+            int serviceOperationResult = _context.Execute<int>(new Uri(_baseUri + "RetrieveProduct", UriKind.Absolute), "POST", true).Single();
+            Assert.Equal(-10, serviceOperationResult);
+
+            // actions
+            int productId = _context.Execute<int>(productOperationDescriptor.Target, "POST", true).Single();
+            Assert.Equal(-10, productId);
+
+            int orderLineId = _context.Execute<int>(orderLineOperationDescriptor.Target, "POST", true).Single();
+            Assert.Equal(-10, orderLineId);
+        }
+
+        [Fact]
+        public void ExecuteOverloadedActionNotBound()
+        {
+            // unbound action
+            _context.Execute(new Uri(_baseUri + "Default.UpdatePersonInfo", UriKind.Absolute), "POST");
+        }
+
+        [Fact]
+        public void ExecuteActionImport()
+        {
+            var productId = _context.RetrieveProduct().GetValue();
+            Assert.Equal(-10, productId);
+            _context.UpdatePersonInfo();
+        }
+
+        [Fact, TestPriority(1)]
+        public void ExcuteBoundAction()
+        {
+            _context.MergeOption = MergeOption.OverwriteChanges;
+
+            Common.Clients.EndToEnd.Employee employee = (Common.Clients.EndToEnd.Employee)_context.People.Where(p => p.PersonId == 0).Single();
+            Assert.Equal(85, employee.Salary);
+
+            DataServiceQuerySingle<Common.Clients.EndToEnd.Employee> singleEmployee = new DataServiceQuerySingle<Common.Clients.EndToEnd.Employee>(_context, "People(0)/Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.Employee");
+            singleEmployee.UpdatePersonInfo().Execute();

```

Comment:
What does this statement do? How is the `singleEmployee` updated? Any assertions?